source/utilities.cc

```diff
@@ -2532,6 +2532,21 @@ namespace aspect
     }
 
 
+    template <int dim>
+    SymmetricTensor<2,dim>
+    symmetric_independent_component_matrix (const unsigned int k)
+    {
+      // somehow I need this, because otherwise it complains that I am passing 3 arguments and it only takes 2...
+      const bool boolian = k < SymmetricTensor<2,dim>::n_independent_components;
```

Comment:
@bangerth: I have tested this and found the same strange behavior. The following code does not work with deal.II 8.5, although it should:
```
const unsigned int k=0;
Assert(k< SymmetricTensor<2,dim>::n_independent_components,ExcMessage(""))
```

The compiler error message is:
```
/home/rengas/Software/Aspect-Versionen/aspect/source/material_model/visco_plastic.cc:815:80: error: macro "Assert" passed 3 arguments, but takes just 2
       Assert(k< SymmetricTensor<2,dim>::n_independent_components,ExcMessage(""));
                                                                                ^
```

Reply:
As discussed on the way to lunch, you'll have to put a set of (parentheses) around the condition or the preprocessor is going to interpret the comma in `2,dim` as one that separates macro arguments.